### DISCUSSION TITLE
The `--all` parameter should definitely be added to the Search Option

### DISCUSSION THREAD

**Discussion by spamkutu** [containers/podman#23376] [Ideas]
podman search registry.access.redhat.com/httpd
Podman list the HTTP images in the Red Hat registry. This is okay.

podman search httpd
Podman list the HTTP images in the default Docker registry. This is good too.

podman search httpd --all
error: `unknown flag: --all`

But I have 3-4 more registered registries, and the `--all` parameter, which is traditionally available in Podman, is not yet defined for the **Search** option. That's why we get the error: `**unknown flag: --all**`. The `--all` parameter should definitely be added to the search option, and it would be very convenient to obtain information from all the registries we have.

**Comment by Luap99** [comment]
That is already the default

> By default, all unqualified-search registries in containers-registries.conf(5) are used.

https://docs.podman.io/en/latest/markdown/podman-search.1.html

**Comment by spamkutu** [comment]
I use repository from ;
"registry.redhat.io", "registry.access.redhat.com", "registry.fedoraproject.org", "docker.io", "ghcr.io", "quay.io" etc etc..

Below, as specified, to exact adresses, So I need to search and check all "my-image" accross all registry repositories with a single command.
"registry.redhat.io/my-repository/my-image, registry.fedoraproject.org/my-repository/my-image, quay.io/my-repository/my-image, ghcr.io/my-repository/my-image, or docker.io/my-repository/my-image"
Is there a comprehensive query available for all existing repositories, rather than querying individual repositories one by one? When I looked at the man page, I didn't see a `--all` or other parameter for performing a search query across all repositories. 

Maybe a traditional `--all` parameter would work for this; it's very simple, but from what I've seen, it hasn't been included.



"""That is already the default

By default, all unqualified-search registries in containers-registries.conf are used.""""

However, if we do not query each repository individually, the result only comes based on a single repository. At this point, I must correct you; I may be mistaken or incomplete, but all repositories are definitely not listed on my screen in this way."
Is there a comprehensive query available for all existing repositories, rather than querying individual repositories one by one? When I looked at the man page, I didn't see a `--all` or other parameter for performing a search query across all repositories. How will be this done? This is what I intended to express. As output, I only get responses from one respository on my screen.

**Reply by Luap99** [reply]
It will only shows the first 25 results by default. You might need to set --limit to something higher.

This definitely works on the latest version

```
$ podman search testimage
NAME                                                       DESCRIPTION
docker.io/pravega/pravega-operator-testimages              
docker.io/pravega/bookkeeper-testimages                    BookKeeper images used for automated testing
docker.io/pravega/pravega-testimages                       Pravega images used for automated testing
docker.io/sonobuoy/testimage                               
docker.io/testimages2022/nginx1                            
docker.io/testimages2022/multi-arch-arm64-arm7             
docker.io/testimages2022/oci_image_index_armv7_amd64       
docker.io/testimages2022/multi-arch-arm64-amd64            
docker.io/testimages2022/oci_image_index_arm64_armv7       
docker.io/shadowcluster/testimage                          a test ubuntu image
docker.io/gesellix/testimage                               A Testimage used for Docker Client integrati...
docker.io/vulntesting2/testimage                           
docker.io/testimagess/test                                 
docker.io/securesystemsengineering/testimage               
docker.io/acharyasrinivasprasad/testimage                  
docker.io/anujsehgal/testimage                             
docker.io/goodwork2020/testimage                           
docker.io/sanketghule/testimage                            
docker.io/gmaindol/testimage                               testimage for compliances
docker.io/juledwar/testimage                               
docker.io/n3wt0n/dependabot-azuredevops-atscale-testimage  Test image for the Dependabot for Azure DevO...
docker.io/aszepesi/testimage                               
docker.io/testimage/tomcat_hello_world                     tomcat with app
docker.io/vidyar/testimage                                 
docker.io/andxu/testimage                                  
quay.io/libpod/testimage                                   Image used by podman CI testing; see test/sy...
quay.io/snakemake/containerize-testimage                   
quay.io/imtiyaz_otel/testimage                             
quay.io/libpod/buildah-testimage-nohistory                 
quay.io/maratsal/mytestimage                               
quay.io/llasmith/testimage                                 
quay.io/bootc-extension/testimage                          
quay.io/higginsd/testimage                                 
quay.io/sshnaidm1/testimage                                
quay.io/troyer/mlflow-server                               # MLFlow Server  An MLFlow Server designed t...
quay.io/psundara/testimage                                 
quay.io/madorn/testimage                                   
quay.io/enzyme/testimage                                   
quay.io/maximeheckel/testimage                             
quay.io/svidrascu/testimage                                
quay.io/amar24/testimages                                  
quay.io/ctryti/testimage                                   
quay.io/pfhost/testimage                                   
quay.io/pfhost/testimage2                                  
quay.io/chakim/testimage                                   
quay.io/kevinrizza/testimage                               
quay.io/yongiman/testimage1                                
quay.io/test_openstack/gcr.io-testimage                    
quay.io/bgallagher/testimage                               
quay.io/tmckayus/testimage                                 
```

**Reply by spamkutu** [reply]
Its just like what I asked for thanks I will update as soon as possible on Podman Desktop.
